frontend/src/metabase-lib/filter.ts

```diff
@@ -393,7 +399,9 @@ export function excludeDateFilterParts(
 
   const bucket = temporalBucket(column);
   if (!bucket) {
-    return { column, operator, bucket, values: [] };
+    return serializedValues.length === 0
```

Comment:
Bug 2 - without this check we accepted unsupported expressions like "[!=, column, column]" as valid exclude date filters, which is not correct (tested).